test/e2e_node/log_path_test.go

```diff
@@ -142,6 +144,51 @@ var _ = framework.KubeDescribe("ContainerLogPath [NodeConformance]", func() {
 				podClient.Create(checkPod)
 				err = framework.WaitForPodSuccessInNamespace(f.ClientSet, checkPodName, ns)
 				framework.ExpectNoError(err, "Failed waiting for pod: %s to enter success state", checkPodName)
+
+				podID := string(createdLogPod.UID)
+				expectedCRILogFile := logCRIDir + "/" + podID + "/" + logContName + "/0.log"
+
+				checkCRIPod := &v1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: checkCRIPodName,
+					},
+					Spec: v1.PodSpec{
+						// this pod is expected to exit successfully
+						RestartPolicy: v1.RestartPolicyNever,
+						Containers: []v1.Container{
+							{
+								Image: busyboxImage,
+								Name:  checkCRIContName,
+								// If we find expected cri log file and contains right content, exit 0
+								// else, keep checking until test timeout
+								Command: []string{"sh", "-c", "while true; do if [ -e " + expectedCRILogFile + " ] && grep -q " + logString + " " + expectedCRILogFile + "; then exit 0; fi; sleep 1; done"},
```

Comment:
maybe it's better to wait for pod success and check the log file afterward?

Reply:
/cc @feiskyer

Reply:
I see. Thanks for the explanation.